src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -2002,9 +2002,8 @@ Well{0} entered with 'FIELD' parent group:
             const auto well_names = this->wellNames(wellNamePattern, handlerContext);
             for (const auto& well_name: well_names) {
                 auto well = this->snapshots.back().wells(well_name);
-                const auto filter_conc = record.getItem<ParserKeywords::WINJFCNC::VOL_CONCENTRATION>().get<double>(0);
-                // the unit is ppm_vol
-                well.setFilterConc(filter_conc/1.e6);
+                const auto filter_conc = record.getItem<ParserKeywords::WINJFCNC::VOL_CONCENTRATION>().get<UDAValue>(0);
+                well.setFilterConc(filter_conc );
```

Comment:
What happened to the factor of `1.0e6`?  I don't see this in the unit handling.

Reply:
I moved to scale in the simulator side, https://github.com/OPM/opm-simulators/pull/4763/files#diff-27ec506a141c179a9728b3453b16c0df44e43b734900be39b2a4709fc0ee4316L71 . 

```c++
filtration_particle_volume_[perf] += water_rates * (conc * 1.e-6) * dt; // conc is with ppm_vol unit
```

Basically I really do not know how we should handle the PPM volume unit properly due to the limited view of the all the components. 

Will a dimension `1.e-6` in the json file do anything better in this direction. 
```
    {
      "name": "VOL_CONCENTRATION",
      "value_type": "DOUBLE",
      "dimension": "1e-6",
      "default" : 0
    }
```
And also, should I use the `Dimension` in the class `UDAValue` to make things better presented?  Suggestions on these will be much appreciated.

Reply:
> Will a dimension `1.e-6` in the json file do anything better in this direction.

No, not really.  I would actually hope you'd get a build failure if you tried to write that in the JSON file.  The only permissible `"dimension"` strings are supposed to be (combinations of) the dimensions explicitly registered/added in [`UnitSystem.cpp`](https://github.com/OPM/opm-common/blob/0895983ac1ccc6dc6aa2bde88a047419327d97e4/src/opm/input/eclipse/Units/UnitSystem.cpp#L1181-L1207).  Maybe we could
```C++
addDimension("PPM", 1.0e-6);
```
or something to that effect in all unit conventions and then switch the JSON-file to using `"dimension" : "PPM"`?

> And also, should I use the `Dimension` in the class `UDAValue` to make things better presented?

If we do `addDimension("PPM", 1.0e-6)`, then we should definitely use `UDAValue::get_dim().convertRawToSi()` to compute the concentration value.  That is more or less exactly how we compute target/limit values in `WCON*` and `GCON*` that are given by UDAs.  As an example, please have a look at [`OPM::UDA::eval_well_uda()`](https://github.com/OPM/opm-common/blob/0895983ac1ccc6dc6aa2bde88a047419327d97e4/src/opm/input/eclipse/Schedule/eval_uda.cpp#L47).

Reply:
Thanks. I will try something along this line and update the PR.

Reply:
Now I use the newly introduced `PPM` unit. It works well.